### PR TITLE
Fix retired swapchain image deletion

### PR DESCRIPTION
**Pull request by hevrard**
The deletion of retired swapchain images could actually remove a
non-swapchain image from the global Images map that tracks Vulkan
images as part of the Vulkan state.

Consider these commands, and how our code was working before this
change:

```
vkCreateSwapchainKHR(oldSwapchain:0x123) # retires swapchain 0x123
                                         # which has img 0x456
					 # that is not acquired

vkCreateImage() # Returns a fresh new image with handle 0x456

vkDestroySwapchainKHR(0x123) # This removes img handle 0x456 from
                             # Images, and that is incorrect
```

This change fixes how retired swapchain images are handled, and
properly deletes them from Images only when it is relevant. See
code comments for details.

Bug: b/190829522
Test: manual, on an app that systematically triggered an error.

### CODE REVIEW

**Review by matled — COMMENTED** [review]
LGTM

Thanks for the detailed PR description and comments!

Not approving as I don't have much context.

File: gapis/api/vulkan/extensions/khr_swapchain.api [hunk 1]
```diff
@@ -193,7 +205,7 @@ sub bool imageUsedByAnotherSwapchain(
     for _, _, s in Swapchains {
       if (s != null) && (s != swapchainObject) {
         for _, _, i in s.SwapchainImages {
-          if i == imageObject {
+          if i.VulkanHandle == imageObject.VulkanHandle {
```
Comment (yalcinmelihyasin) [inline gapis/api/vulkan/extensions/khr_swapchain.api:208]: What is the reason for this change?

**Review by yalcinmelihyasin — APPROVED** [review]
LGTM

Only one minor question.

File: gapis/api/vulkan/extensions/khr_swapchain.api (hunk 1 above)
Comment (hevrard) [reply gapis/api/vulkan/extensions/khr_swapchain.api:208]: The code used to do a "deep equality" test, whereas we only want to check whether the handles are the same: this is particularly important as if an image handle from a retired swapchain image has been "recycled" for another swapchain image, then the image objects may have the same Vulkan handle, but be different otherwise.

Comment (yalcinmelihyasin) [reply gapis/api/vulkan/extensions/khr_swapchain.api:208]: I see. Makes sense.

**Review by ianelliottus — APPROVED** [review]
I'm not familiar with this code, but I've tried to understand what it's doing and it LGTM
